src/client/datascience/notebook/kernelWithMetadata.ts

```diff
@@ -63,6 +63,13 @@ export class VSCodeNotebookKernelMetadata implements VSCNotebookKernel {
      * resolved, the cell will be put back into the Idle state.
      */
     public async executeCellsRequest(document: NotebookDocument, ranges: NotebookCellRange[]): Promise<void> {
+        // When we receive a cell execute request, first ensure that the notebook is trusted.
+        // If it isn't already trusted, block execution until the user trusts it.
+        const isTrusted = await this.commandManager.executeCommand(Commands.TrustNotebook, document.uri);
```

Comment:
i'd first check if the notebook is already trusted. if not trusted, then perform this async operation.

Reply:
```
        const model = this.storageProvider.get(document.uri);
        if (model && model.isTrusted === false) {
			const isTrusted = await this.commandManager.executeCommand(Commands.TrustNotebook, document.uri);
```

Else calling a method to trust a notebook when its already trusted seems wrong.

Reply:
I'm not sure I understand why that's necessary. This command already checks if the model is trusted and returns true if it is, so we'd be performing the same check twice if the document wasn't trusted. This would also necessitate injecting the NotebookStorageProvider into the VSCNotebookKernel, which seems like a weird dependency for VSCNotebookKernel to have to take.

Reply:
> his command already checks if the model is trusted and returns true if it is, so we'd be performing the same check twice if the document wasn't trusted

I don't like the idea of calling a command `trustNotebook` when its already trusted. We should call methods `TrustNotebook` only if something is not trusted.

But, now i see that this is not somethign I should block the PR on. hence approving (as its just a personal preference).

Reply:
Yeah the command name threw me too. Perhaps it should be renamed to `Commands.PromptToTrustNotebook`?

Reply:
I've leave it as is for now, this will go away soon, when VS Code has trust.